### PR TITLE
upgrade RESTEasy and Jakarta Spec version

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,11 +22,12 @@
         <maven.test.redirectTestOutputToFile>true</maven.test.redirectTestOutputToFile>
         <!-- Dependency versions, please keep in alphabetical order -->
         <version.jakarta.transaction.transaction-api>1.3.3</version.jakarta.transaction.transaction-api>
-        <version.javax.ws.rs-api>2.0.1.Final</version.javax.ws.rs-api>
+        <version.jakarta.ws.rs-api>3.1.0.RC1-jbossorg-1</version.jakarta.ws.rs-api>
```
Comment (liweinan) [inline pom.xml:25]: @jamezp The build failed with:

```
Could not find artifact jakarta.ws.rs:jakarta.ws.rs-api:jar:3.1.0.RC1-jbossorg-1 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

How could I fix this(setup proper repo)?

Comment (liweinan) [reply pom.xml:25]: I added the jboss repo and this issue fixed.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -18,21 +18,21 @@ jobs:
...
-       run: mvn -U -B clean install
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
```
Comment (jamezp) [inline .github/workflows/maven.yml:32]: This should change to `uses: actions/setup-java@2`, then add `cache: 'maven'`. We can then remove the full `-uses: actions/cache@v1`.

**Review by jamezp — APPROVED** [review]

File: .github/workflows/maven.yml (hunk 2 above)
Comment (liweinan) [reply .github/workflows/maven.yml:32]: @jamezp Thanks for working on this!

**Pull request merged**